girder/api/v1/assetstore.py

```diff
@@ -248,3 +250,96 @@ def getAssetstoreFiles(self, assetstore, limit, offset, sort, params):
         return list(self.model('file').find(
             query={'assetstoreId': assetstore['_id']},
             offset=offset, limit=limit, sort=sort))
+
+    @access.admin
+    @autoDescribeRoute(
+        Description('Get statistics by collection and by user.')
+        .modelParam('id', model='assetstore')
+        .errorResponse()
+        .errorResponse('You are not an administrator.', 403)
+    )
+    def getStats(self, assetstore, params):
+        filesInStore = list(self.model('file').find(
+            query={'assetstoreId': assetstore['_id']}, fields=['size', 'creatorId', 'itemId']))
+
+        collectionItemsInStore = list(self.model('item').find(query={
+            '_id': {
+                '$in': list(set([x['itemId'] for x in filesInStore]))
+            },
+            'baseParentType': 'collection'
+        }))
+        collectionItemsInStoreMap = dict((x['_id'], x) for x in collectionItemsInStore)
+
+        collectionUsingStore = list(self.model('collection').find(
+            query={
+                '_id': {
+                    '$in': list(set([x['baseParentId'] for x in collectionItemsInStore]))
+                }
+            }))
+        collectionUsingStoreMap = dict((x['_id'], x) for x in collectionUsingStore)
+
+        collectionFileMap = {}
+        for file in filesInStore:
+            if file['itemId'] in collectionItemsInStoreMap:
+                collectionId = collectionItemsInStoreMap[file['itemId']]['baseParentId']
+                if collectionId not in collectionFileMap:
+                    collectionFileMap[collectionId] = [file]
+                else:
+                    collectionFileMap[collectionId].append(file)
+
+        usersUsingStore = list(self.model('user').find(
+            query={
+                '_id': {
+                    '$in': list(set([x['creatorId'] for x in filesInStore]))
+                }
+            }))
+        userUsingStoreMap = dict((x['_id'], x) for x in usersUsingStore)
+        userFileMap = {}
+        for file in filesInStore:
+            userId = file['creatorId']
+            if userId not in userFileMap:
+                userFileMap[userId] = [file]
+            else:
+                userFileMap[userId].append(file)
+
+        # based on collected data stored in dictionary, aggregate size and format
+        # result. Ignore the fact there will be only one copy for the same file
+        # in the assetstore.
+        currentUser = self.getCurrentUser()
```

Comment:
Per my comments below, I don't think the output should depend on the user calling this endpoint (since it's admin-only).